### PR TITLE
Buffer key enums

### PR DESCRIPTION
**Pull request by chriselion**
### Proposed change(s)

Describe the changes made in this PR.

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: ml-agents/mlagents/trainers/buffer.py [hunk 1]
```diff
@@ -14,144 +18,235 @@ class BufferException(UnityException):
...
+    @staticmethod
+    def returns_key(name: str) -> AgentBufferKey:
+        return RewardSignalKeyPrefix.RETURNS, name
+
+    @staticmethod
+    def advantage_key(name: str) -> AgentBufferKey:
+        return RewardSignalKeyPrefix.ADVANTAGE, name
+
+
+class AgentBufferField(list):
```
Comment (chriselion) [inline ml-agents/mlagents/trainers/buffer.py:73]: This was un-nested from AgentBuffer, no other changes.

File: ml-agents/mlagents/trainers/buffer.py [hunk 2]
```diff
@@ -14,169 +18,309 @@ class BufferException(UnityException):
...
+    RETURNS = "returns"
+    ADVANTAGE = "advantage"
+
+
+AgentBufferKey = Union[
+    BufferKey, Tuple[ObservationKeyPrefix, int], Tuple[RewardSignalKeyPrefix, str]
+]
+
+
+class RewardUtil:
```
Comment (ervteng) [inline ml-agents/mlagents/trainers/buffer.py:55]: Really small point: what do you think about renaming this class `RewardSignalUtil`? To align with `RewardSignalKeyPrefix` and separating it from the rewards in the buffer

**Review by ervteng — APPROVED** [review]
👍

File: ml-agents/mlagents/trainers/buffer.py (hunk 2 above)
Comment (chriselion) [reply ml-agents/mlagents/trainers/buffer.py]: Sure, was just trying to avoid a few line breaks. Updated.

**Pull request merged**